include/globals.h

```diff
@@ -12,26 +12,32 @@ extern byte settings;
 extern byte aftertouch;
 extern bool aftertouchToLfo;
 extern bool sendLfo;
+extern bool volumeChanged;
 extern bool sendArp;
 extern bool lfoNewRand[3];
 extern float bend, bend1, bend2, bend3;
 extern bool sync;
+extern bool pwLimit;
+extern bool toolMode;
 extern bool cvActive[3];
 extern int lfoStep[3];
 extern byte lfo[3];
 extern int velocityToLfo;
 extern bool modToLfo;
 extern byte velocityLast;
 extern byte modWheelLast;
-
+extern bool filterModeHeldGlobal;
```

Comment:
Don't introduce that duplicate variable here. It's already in `UiState` which is defined in `ui_vars.h`. There is a reason for it to be there, and whenever you do such a thing, you should ask yourself first: "What was the reason it was designed that way in the first place, and why do I need to work around this design now?".

In this case, the answers are:

- Reason: It was designed that way to keep the `filterModeHeld` variable away from all the other code which is not UI. Because the voice allocation, the preset load/save, the arpeggiator etc should never care about UI details like this. And the best way to ensure that they really don't care, is to not give them this variable into (global) scope in the first place. (This helps keeping your code nicely decoupled and maintainable).
- Why work around: I'm not sure tbh. Probably because the `ui_vars.h` wasn't included in `ui_pots.cpp`. However, since it's literally UI handling code, it just feels right to add that include there. (It would feel wrong to do the same in `sid.cpp` or `arp.cpp`).